lib/AppInfo/DashboardWidget.php

```diff
@@ -61,4 +70,29 @@ public function getUrl(): ?string {
 	public function load(): void {
 		\OCP\Util::addScript('notes', 'notes-dashboard');
 	}
+
+	public function getWidgetButtons(string $userId): array {
+		$buttons = [
+			new WidgetButton(WidgetButton::TYPE_NEW, $this->url->linkToRouteAbsolute('notes.page.create'), $this->l10n->t('Create new note'))
+		];
+		if ($this->notesService->countNotes($userId) > 7) {
+			$buttons[] = new WidgetButton(WidgetButton::TYPE_MORE, $this->url->linkToRouteAbsolute('notes.page.index'), $this->l10n->t('More notes'));
+		}
+		return $buttons;
+	}
+
+	public function getItems(string $userId, ?string $since = null, int $limit = 7): array {
+		$notes = $this->notesService->getTopNotes($userId);
+		$notes = array_slice($notes, 0, $limit);
+		return array_map(function (Note $note) {
+			$excerpt = '';
+			try {
+				$excerpt = $note->getExcerpt();
+			} catch (\Throwable $e) {
+			}
+			$link = $this->url->linkToRouteAbsolute('notes.page.index', ['id' => $note->getId()]);
```

Comment:
Is this correct link?
It is something like: https://cloud.com/index.php/apps/notes/?id=271
but on web I see:
https://cloud.com/apps/notes/note/6004520

Thus all links via widget end up showing any previous note but not the desired one.

Reply:
Yep, it must be `notes.page.indexnote` instead:

```suggestion
			$link = $this->url->linkToRouteAbsolute('notes.page.indexnote', ['id' => $note->getId()]);
```

See e.g. https://github.com/nextcloud/notes/blob/v4.5.0/lib/Controller/PageController.php#L62